components/cam/src/physics/crm/crm_physics.F90

```diff
@@ -813,12 +837,80 @@ subroutine crm_physics_tend(ztodt, state, tend, ptend, pbuf, cam_in, cam_out, &
+      if (masterproc) then
+        if (use_crm_accel) then
+#if !defined(sam1mom)
+          write(0,*) "CRM time step relaxation is only compatible with sam1mom microphysics"
+          call endrun('crm main')
+#endif
+        endif
+      endif
+
+      ! Load the nstep
+      igstep = get_nstep()
+
+#ifdef MMF_SAM
+
       call t_startf ('crm_call')
-      call crm( lchnk, ncol, ztodt, pver,       &
-                crm_input, crm_state, crm_rad,  &
-                crm_ecpp_output, crm_output )
+
+      call crm(lchnk, ncol, ztodt, pver, crm_input, crm_state, crm_rad, crm_ecpp_output, crm_output, &
+               latitude0, longitude0, gcolp, igstep, use_crm_accel_tmp, crm_accel_factor, crm_accel_uv_tmp)
+
       call t_stopf('crm_call')
 
+#elif defined(MMF_SAMXX)
+
+      call t_startf ('crm_call')
+
+      ! Fortran classes don't translate to C++ classes, we we have to separate
+      ! this stuff out when calling the C++ routinte crm(...)
+      call crm(ncol, pcols, ztodt, pver, crm_input%bflxls, crm_input%wndls, crm_input%zmid, crm_input%zint, &
+               crm_input%pmid, crm_input%pint, crm_input%pdel, crm_input%ul, crm_input%vl, &
+               crm_input%tl, crm_input%qccl, crm_input%qiil, crm_input%ql, crm_input%tau00, &
+               crm_state%u_wind, crm_state%v_wind, crm_state%w_wind, crm_state%temperature, &
+               crm_state%qt, crm_state%qp, crm_state%qn, crm_rad%qrad, crm_rad%temperature, &
+               crm_rad%qv, crm_rad%qc, crm_rad%qi, crm_rad%cld, crm_output%timing_factor, &
+               crm_output%prectend, crm_output%precstend, crm_output%cld, crm_output%cldtop, &
+               crm_output%gicewp, crm_output%gliqwp, crm_output%mctot, crm_output%mcup, crm_output%mcdn, &
+               crm_output%mcuup, crm_output%mcudn, crm_output%qc_mean, crm_output%qi_mean, crm_output%qs_mean, &
+               crm_output%qg_mean, crm_output%qr_mean, crm_output%mu_crm, crm_output%md_crm, crm_output%eu_crm, &
+               crm_output%du_crm, crm_output%ed_crm, crm_output%flux_qt, crm_output%flux_u, crm_output%flux_v, &
+               crm_output%fluxsgs_qt, crm_output%tkez, crm_output%tkesgsz, crm_output%tkz, crm_output%flux_qp, &
+               crm_output%precflux, crm_output%qt_trans, crm_output%qp_trans, crm_output%qp_fall, crm_output%qp_evp, &
+               crm_output%qp_src, crm_output%qt_ls, crm_output%t_ls, crm_output%jt_crm, crm_output%mx_crm, crm_output%cltot, &
+               crm_output%clhgh, crm_output%clmed, crm_output%cllow, crm_output%sltend, crm_output%qltend, crm_output%qcltend, &
+               crm_output%qiltend, crm_output%tk, crm_output%tkh, crm_output%qcl, crm_output%qci, crm_output%qpl, crm_output%qpi, &
+               crm_output%z0m, crm_output%taux, crm_output%tauy, crm_output%precc, crm_output%precl, crm_output%precsc, &
+               crm_output%precsl, crm_output%prec_crm, latitude0, longitude0, gcolp, igstep, &
+               use_crm_accel, crm_accel_factor, crm_accel_uv)
```

Comment:
Should we just make these interfaces identical now instead of dealing with it later?
I'll have to change the interface when I address the aerosol optics stuff, so I can take care of it then.

Reply:
Changing it now will mess with @xyuan 's upcoming PR. I'd rather wait until after his PR to do this.

Reply:
ok, well the changes I need to make will probably mess with @xyuan 's PR either way then.

Reply:
He said he's OK with us changing this. I can make them the same. I'll work on it today.

Reply:
Actually, @whannah1 , this is going to get complicated. `crm_output_module.F90` has ifdefs for M2005, ESMT, and MODAL_AERO. Those were pretty ugly in the interface before. They also aren't currently passed to samxx because they aren't used. This might get pretty messy.

Reply:
ok, then let's just leave it and figure it out later. My hope is that we can implement P3 in a way that doesn't require any ifdef nonsense, but maybe after that's implemented we can just delete the fortran SAM code and solve this problem.